server/routes_test.go

```diff
@@ -53,6 +53,40 @@ func TestRouteConfig(t *testing.T) {
 	}
 }
 
+func TestRouteAdvertise(t *testing.T) {
+	// TODO: Need to work through this test case. may need to add a util proxy server
+	// to validate functionally.
+	optsSeed, _ := ProcessConfigFile("./configs/seed.conf")
+
+	optsSeed.NoSigs, optsSeed.NoLog = true, false
+	optsSeed.Debug = true
+
+	srvSeed := RunServer(optsSeed)
+	defer srvSeed.Shutdown()
+
+	seedRouteUrl := fmt.Sprintf("nats://%s:%d", optsSeed.Cluster.Host,
+		srvSeed.ClusterAddr().Port)
+	optsA := nextServerOpts(optsSeed)
+	optsA.Routes = RoutesFromStr(seedRouteUrl)
+	optsA.Cluster.Port = 9999
+	optsA.Cluster.RouteAdvertise = "example.com:80"
+
+	srvA := RunServer(optsA)
+	defer srvA.Shutdown()
+
+	if srvA.routeInfo.Host != "example.com" {
```

Comment:
You are accessing internals here, make sure that locking is ok (to avoid race reports).

Reply:
What I usually do is capture the info under lock and then test/fail outside of the lock.
For instance:
```
srvA.mu.Lock()
h := srvA.routeInfo.Host
ip := srvA.routeInfo.IP
srvA.mu.Unlock()
if h != ...
```